cs_unittest/TestModelLoading.cs

```diff
@@ -10,41 +10,55 @@
 namespace cs_unittest
 {
     [TestClass]
-    [Ignore]
     public class TestModelLoading
     {
         [TestMethod]
         [DeploymentItem(@"model-sets\7.10.2_corrupted.model", "model-sets")]
         [DeploymentItem(@"model-sets\8.0.0_ok.model", "model-sets")]
+        [DeploymentItem(@"model-sets\8.0.1_rcv1_ok.model", "model-sets")]
+        [DeploymentItem(@"model-sets\8.0.1.test_named_ok.model", "model-sets")]
+        [DeploymentItem(@"model-sets\8.0.1_hash_ok.model", "model-sets")]
         public void TestLoadModel()
         {
             InternalTestModel(@"model-sets/7.10.2_corrupted.model", false);
             InternalTestModel(@"model-sets/8.0.0_ok.model", true);
+            InternalTestModel(@"model-sets/8.0.1.test_named_ok.model", true);
+            InternalTestModel(@"model-sets/8.0.1_rcv1_ok.model", true);
+            InternalTestModel(@"model-sets/8.0.1_hash_ok.model", true);
         }
 
         [TestMethod]
-        [DeploymentItem(@"model-sets\rcv1.model", "model-sets")]
-        [DeploymentItem(@"model-sets\test_named.model", "model-sets")]
+        [DeploymentItem(@"model-sets\8.0.1_rcv1_ok.model", "model-sets")]
+        [DeploymentItem(@"model-sets\8.0.1.test_named_ok.model", "model-sets")]
+        [DeploymentItem(@"model-sets\8.0.1_hash_ok.model", "model-sets")]
         public void TestLoadModelRandomCorrupt()
         {
-            InternalTestModelRandomCorrupt("model-sets/test_named.model");
-            InternalTestModelRandomCorrupt("model-sets/rcv1.model");
+            InternalTestModelRandomCorrupt("model-sets/8.0.1.test_named_ok.model");
+            InternalTestModelRandomCorrupt("model-sets/8.0.1_rcv1_ok.model");
+            InternalTestModelRandomCorrupt("model-sets/8.0.1_hash_ok.model");
         }
 
         private void InternalTestModel(string modelFile, bool shouldPass)
         {
+            bool passed = false;
             try
             {
                 using (var vw = new VowpalWabbitModel(string.Format("--quiet -t -i {0}", modelFile)))
                 {
                     // should only reach this point if model is valid
-                    Assert.IsTrue(shouldPass);
+                    passed = true;
                 }
             }
             catch (VowpalWabbitException ex)
             {
                 Assert.IsTrue(ex.Message.Contains("corrupted"));

```

Comment:
Looks neat, although I wanted to inspect the exception message before asserting. I guess the ideal thing would be to throw specific exception type and use the annotation here, but I don't think that is possible from C++ side, so we'd need to catch in CLI, inspect message, and rethrow, which seems messy...